contracts/src/libraries/PlonkVerifier.sol

```diff
@@ -171,84 +168,241 @@ library PlonkVerifier {
+            mstore(add(dataPtr, 0x520), mload(add(sigma3Ptr, 0x20))) // sigma3.y
+            let sigma4Ptr := mload(add(vk, 0xc0))
+            mstore(add(dataPtr, 0x540), mload(sigma4Ptr)) // sigma4.x
+            mstore(add(dataPtr, 0x560), mload(add(sigma4Ptr, 0x20))) // sigma4.y
+
+            // public inputs
+            // TODO: update these when change PI from dynamic array to fixed length
+            mstore(add(dataPtr, 0x580), mload(add(pi, 0x20))) // PI[0]
+            mstore(add(dataPtr, 0x5a0), mload(add(pi, 0x40))) // PI[1]
+            mstore(add(dataPtr, 0x5c0), mload(add(pi, 0x60))) // PI[2]
+            mstore(add(dataPtr, 0x5e0), mload(add(pi, 0x80))) // PI[3]
+            mstore(add(dataPtr, 0x600), mload(add(pi, 0xa0))) // PI[4]
+            mstore(add(dataPtr, 0x620), mload(add(pi, 0xc0))) // PI[5]
+            mstore(add(dataPtr, 0x640), mload(add(pi, 0xe0))) // PI[6]
+            mstore(add(dataPtr, 0x660), mload(add(pi, 0x100))) // PI[7]
+
+            // proof
+            let wire0Ptr := mload(proof)
+            mstore(add(dataPtr, 0x680), mload(wire0Ptr)) // wire0.x
+            mstore(add(dataPtr, 0x6a0), mload(add(wire0Ptr, 0x20))) // wire0.y
+            let wire1Ptr := mload(add(proof, 0x20))
+            mstore(add(dataPtr, 0x6c0), mload(wire1Ptr)) // wire1.x
+            mstore(add(dataPtr, 0x6e0), mload(add(wire1Ptr, 0x20))) // wire1.y
+            let wire2Ptr := mload(add(proof, 0x40))
+            mstore(add(dataPtr, 0x700), mload(wire2Ptr)) // wire2.x
+            mstore(add(dataPtr, 0x720), mload(add(wire2Ptr, 0x20))) // wire2.y
+            let wire3Ptr := mload(add(proof, 0x60))
+            mstore(add(dataPtr, 0x740), mload(wire3Ptr)) // wire3.x
+            mstore(add(dataPtr, 0x760), mload(add(wire3Ptr, 0x20))) // wire3.y
+            let wire4Ptr := mload(add(proof, 0x80))
+            mstore(add(dataPtr, 0x780), mload(wire4Ptr)) // wire4.x
+            mstore(add(dataPtr, 0x7a0), mload(add(wire4Ptr, 0x20))) // wire4.y
+
+            // challenge: beta
+            {
+                mstore(statePtr, 0x0) // init state
+                // preimage len: state(0x20) + transcript(0x7c0)
+                mstore(add(dataPtr, 0x7c0), keccak256(statePtr, 0x7e0))
+                // update new state (by updating state pointer)
+                statePtr := add(dataPtr, 0x7c0)
+                // empty transcript
+                dataPtr := add(statePtr, 0x20)
+                // (mod p) to get beta
+                mstore(add(res, 0x60), mod(mload(statePtr), p))
+            }
 
-        res.zeta = transcript.getChallenge();
+            // challenge: gamma
+            {
+                // preimage len: state(0x20) + transcript(0x0)
+                mstore(dataPtr, keccak256(statePtr, 0x20))
+                // update new state (by updating state pointer)
+                statePtr := dataPtr
+                dataPtr := add(statePtr, 0x20)
```

Comment:
Add comment `// empty transcript` to be consistent with the code above?

Reply:
add in [5f74f10](https://github.com/EspressoSystems/espresso-sequencer/pull/1831/commits/5f74f10918a3fc28a5b53c034e342215abe4cb07) 
similar for all following comments.